blockchain/contracts/Staking.txt

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/interfaces/IERC721.sol";
+import "@openzeppelin/contracts/interfaces/IERC20.sol";
+
+contract Staking {
+    using IERC721 for IERC721;
+    using IERC20 for IERC20;
+    
+    IERC721 public nft;
```

Comment:
Can do this now or later but the name of this variable should be more descriptive - consider `characterToken`